custom/icds_reports/utils/aggregation_helpers/distributed/child_health_monthly.py

```diff
@@ -98,25 +98,43 @@ def _state_aggregation_query(self, state_id):
             "(({} - child_health.opened_on::date)::integer >= 0) "
             "AND (child_health.closed = 0 OR (child_health.closed_on::date - {})::integer > 0)"
         ).format(end_month_string, start_month_string)
+        open_in_month_end = (
+            "(({} - child_health.opened_on::date)::integer >= 0) "
+            "AND (child_health.closed = 0 OR (child_health.closed_on::date - {})::integer > 0)"
+        ).format(end_month_string, end_month_string)
         alive_in_month = "(child_health.date_death IS NULL OR child_health.date_death - {} >= 0)".format(
             start_month_string
         )
+        alive_in_month_end = "(child_health.date_death IS NULL OR child_health.date_death - {} >= 0)".format(
+            end_month_string
+        )
         not_migrated = (
             "(agg_migration.is_migrated IS DISTINCT FROM 1 "
             "OR agg_migration.migration_date::date >= {start_month_string})"
         ).format(start_month_string=start_month_string)
+        not_migrated_month_end = (
+            "(agg_migration.is_migrated IS DISTINCT FROM 1)"
+        )
         registered = (
             "(agg_availing.is_registered IS DISTINCT FROM 0 "
             "OR agg_availing.registration_date::date >= {start_month_string})"
         ).format(start_month_string=start_month_string)
+        registered_month_end = (
+            "(agg_availing.is_registered IS DISTINCT FROM 0 )"
+        )
         seeking_services = "({registered} AND {not_migrated})".format(
             registered=registered, not_migrated=not_migrated)
+        seeking_services_month_end = "({registered_month_end} AND {not_migrated_month_end})".format(
+            registered_month_end=registered_month_end, not_migrated_month_end=not_migrated_month_end)
         born_in_month = "({} AND person_cases.dob BETWEEN {} AND {})".format(
             seeking_services, start_month_string, end_month_string
         )
         valid_in_month = "({} AND {} AND {} AND {} <= 72)".format(
             open_in_month, alive_in_month, seeking_services, age_in_months
         )
+        valid_in_month_end = "({} AND {} AND {} AND {} <= 72)".format(
+            open_in_month_end, alive_in_month_end, seeking_services_month_end, age_in_months_end
```

Comment:
why does it uses age_in_month_end?

Reply:
i think it should be age_in_months right